src/objects.c

```diff
@@ -591,7 +586,25 @@ CK_RV p11prov_obj_find(P11PROV_CTX *provctx, P11PROV_SESSION *session,
         P11PROV_raise(provctx, ret, "Failed to terminate object search");
     }
 
+    for (CK_ULONG k = 0; k < total; k++) {
+        P11PROV_OBJ *obj = NULL;
+        ret = p11prov_obj_from_handle(provctx, session, objects[k], &obj);
+        if (ret == CKR_CANCEL) {
+            /* unknown object or other recoverable error to ignore */
+            continue;
+        }
+        if (ret == CKR_OK) {
+            ret = cb(cb_ctx, obj);
+        }
+        if (ret != CKR_OK) {
```

Comment:
this could use `else if` and `else` respectively, which might make the code slightly less complex.

Reply:
can't do a final else as the last check catches an error in either p11prov_obj_from_handle() or the cb() call. But I can do and if/else if for tje first two